kv/union_store.go

```diff
@@ -127,7 +127,7 @@ func (us *UnionStore) CheckLazyConditionPairs() error {
 	for it := us.lazyConditionPairs.NewIterator(nil); it.Valid(); it.Next() {
 		if len(it.Value()) == 0 {
 			if _, exist := values[it.Key()]; exist {
-				return errors.Trace(ErrKeyExists)
```

Comment:
@ElvizLai 
There is a mechanism in TiDB to raise a predefined error with a particular message. Terror package do this job. You can reference UnknownSystemVar as example. 
To define a new error, we need to do the following work:
1. Create a new Error (such as KeyExists) in terror.go with a class (in this case ClassKV is suitable) and define a new error code in the class for this error.
2. Raise the error and add error message like: 
return terror.KeyExists.Gen("Duplicate key %s", key), this will clone a new KeyExists error object and add message in it.

I will write a document for the error system in TiDB. It is not easy to understand now.